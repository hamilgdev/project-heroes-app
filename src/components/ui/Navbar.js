import { useContext } from "react";
import { Link, NavLink, useNavigate } from "react-router-dom";
import { AuthContext } from "../../auth/authContext";
import { types } from "../../types";

const Navbar = () => {
  const { user, dispatch } = useContext(AuthContext);
  const navigate = useNavigate();

  const handleLogout = () => {
    const action = {
      type: types.logout,
    };
    dispatch(action);
    navigate("/login", { replace: true });
  };

  return (
    <nav className="flex justify-between w-full py-2 px-4 | bg-slate-800 text-slate-400">
      <ul className="flex items-center gap-4">
        <li>
          <Link to="/" className="text-white">
            Asoiciaciones
          </Link>
        </li>
        <li>
          <NavLink
            to="/marvel"
            className={({ isActive }) => (isActive ? "text-cyan-500" : "")}
          >
            Marvel
          </NavLink>
        </li>
        <li>
          <NavLink
            to="/dc"
            className={({ isActive }) => (isActive ? "text-cyan-500" : "")}
          >
            DC
          </NavLink>
        </li>
        <li>
          <NavLink
            to="/search"
            className={({ isActive }) => (isActive ? "text-cyan-500" : "")}
          >
            Search
          </NavLink>
        </li>
      </ul>
      <div className="flex items-center gap-4">
        <h3 className="text-cyan-500 font-semibold">{user.name}</h3>
        <button
          className="px-2 py-1 | bg-sky-900 text-sky-300 | rounded-md | hover:bg-sky-300 hover:text-sky-900"
          onClick={handleLogout}
        >
          Logout
        </button>
      </div>
    </nav>
  );
};

export default Navbar;
