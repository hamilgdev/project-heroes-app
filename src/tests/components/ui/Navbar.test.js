import { mount } from 'enzyme';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { AuthContext } from '../../../auth/authContext';
import { types } from '../../../types';

import Navbar from '../../../components/ui/Navbar';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate
}));

describe('Tests in <Navbar />', () => {
  const contextValue = {
    dispatch: jest.fn(),
    user: { logged: true, name: 'Welcome' }
  };

  const wrapper = mount(
    <AuthContext.Provider value={contextValue}>
      <MemoryRouter initialEntries={['/']}>
        <Routes>
          <Route path='/' element={<Navbar />} />
        </Routes>
      </MemoryRouter>
    </AuthContext.Provider>
  );

  test('should display correctly', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h3').text().trim()).toBe('Welcome');
  });

  test('should call to logout, call the navigate and the dispatch with args', () => {
    wrapper.find('button').prop('onClick')();
    expect(contextValue.dispatch).toHaveBeenCalledWith({ type: types.logout });
    expect(mockNavigate).toHaveBeenCalledWith('/login', { replace: true });
  });
});
