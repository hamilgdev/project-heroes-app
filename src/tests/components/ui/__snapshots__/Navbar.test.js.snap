// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <Navbar /> should display correctly 1`] = `
<nav
  className="flex justify-between w-full py-2 px-4 | bg-slate-800 text-slate-400"
>
  <ul
    className="flex items-center gap-4"
  >
    <li>
      <a
        className="text-white"
        href="/"
        onClick={[Function]}
      >
        Asoiciaciones
      </a>
    </li>
    <li>
      <a
        className=""
        href="/marvel"
        onClick={[Function]}
      >
        Marvel
      </a>
    </li>
    <li>
      <a
        className=""
        href="/dc"
        onClick={[Function]}
      >
        DC
      </a>
    </li>
    <li>
      <a
        className=""
        href="/search"
        onClick={[Function]}
      >
        Search
      </a>
    </li>
  </ul>
  <div
    className="flex items-center gap-4"
  >
    <h3
      className="text-cyan-500 font-semibold"
    >
      Welcome
    </h3>
    <button
      className="px-2 py-1 | bg-sky-900 text-sky-300 | rounded-md | hover:bg-sky-300 hover:text-sky-900"
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
</nav>
`;
